src/libstd/panicking.rs

```diff
@@ -171,8 +173,17 @@ fn default_handler(info: &PanicInfo) {
     let name = thread.as_ref().and_then(|t| t.name()).unwrap_or("<unnamed>");
 
     let write = |err: &mut ::io::Write| {
-        let _ = writeln!(err, "thread '{}' panicked at '{}', {}:{}",
-                         name, msg, file, line);
+        unsafe {
+            if !log_backtrace && FIRST_PANIC.compare_and_swap(true, false, Ordering::SeqCst) {
+                let _ = writeln!(err, "thread '{}' panicked at '{}', {}:{}\n\
+                                       note: Run with `RUST_BACKTRACE=1` for a backtrace.",
+                                 name, msg, file, line);
+            } else {
+                let _ = writeln!(err, "thread '{}' panicked at '{}', {}:{}",
+                                 name, msg, file, line);
```

Comment:
It's fine to have this block be common here and the branch could only surround the `RUST_BACKTRACE` line